bluemira/equilibria/fem_fixed_boundary/equilibrium.py

```diff
@@ -0,0 +1,266 @@
+# License along with bluemira; if not, see <https://www.gnu.org/licenses/>.
+
+"""Fixed boundary equilibrium class"""
+import numpy as np
+
+from bluemira.base.look_and_feel import bluemira_debug, bluemira_print
+from bluemira.equilibria.fem_fixed_boundary.fem_magnetostatic_2D import (
+    FemGradShafranovFixedBoundary,
+)
+from bluemira.equilibria.fem_fixed_boundary.transport_solver import (
+    PlasmodTransportSolver,
+)
+from bluemira.equilibria.fem_fixed_boundary.utilities import (
+    calculate_plasma_shape_params,
+    plot_profile,
+)
+from bluemira.mesh import meshing
+from bluemira.mesh.tools import import_mesh, msh_to_xdmf
+
+
+def solve_plasmod_fixed_boundary(
+    builder_plasma,
+    plasmod_options,
+    gs_options,
+    delta95_t,
+    kappa95_t,
+    lcar_coarse=0.15,
+    lcar_fine=0.05,
+    niter_max=30,
+    iter_err_max=1e-5,
+    theta=0.8,
+    gs_relaxation=0.5,
+    plot=False,
+    verbose=False,
+):
+    """
+    Solve the plasma fixed boundary problem using delta95 and kappa95 as target
+    values and iterating on plasmod to have consistency with pprime and ffprime.
+
+    Parameters
+    ----------
+    builder_plasma: bluemira.base.builder
+        plasma poloidal cross section builder object
+    plasmod_options: dict
+        set of options used to set up and run plasmod
+    gs_options: dict
+        set of options used to set up and run the FemGradShafranovFixedBoundary
+    delta95_t: float
+        target value for delta at 95%
+    kappa95_t: float
+        target value for kappa at 95%
+    lcar_coarse: float
+        value of the characteristic length used to generate the mesh to solve the
+        Grad-Shafranov problem
+    lcar_fine: float
+        value of the characteristic length used to extrapolate the isoflux for the
+        calculation of kappa and delta at 95%
+    niter_max: int
+        maximum number of iteration between Grad-Shafranov and Plasmod
+    iter_err_max: float
+        convergence maximum error to stop the iteration
+    theta: float
+        iteration relaxing factor
+    gs_relaxation: float
+        FemGradShafranovFixedBoundary iteration relaxing factor
+    plot: bool
+        Whether or not to plot
+    verbose: bool
+        Whether or not to print
+
+    Notes
+    -----
+    This function directly modifies the parameters of builder_plasma
+    """
+    niter = 0
+    delta_95 = delta95_t
+    kappa_95 = kappa95_t
+
+    while niter < niter_max:
+        # source string to be used in changed parameters
+        source = f"from equilibrium iteration {niter}"
+
+        # build the plasma
+        # - xz - plasma toroidal cross section
+        # - xyz - to get the volume
+        plasma = builder_plasma.build_xz().get_component("xz").get_component("LCFS")
+        plasma_volume = (
+            builder_plasma.build_xyz()
+            .get_component("xyz")
+            .get_component("LCFS")
+            .shape.volume
+        )
```

Comment:
I can't think when we'd ever use this without cylindrical coordinates, so the CAD here feels a little gratuitous!

What about:
```python
plasma = builder_plasma.build_xz().get_component("xz").get_component("LCFS")
plasma_volume = 2 * np.pi * plasma.shape.center_of_mass[0] * plasma.shape.area
```

Reply:
Done